.github/workflows/haskell.yml

```diff
@@ -1,127 +1,141 @@
-name: Binaries
-
-defaults:
-  run:
-    shell: bash
+name: Haskell CI
 
 on:
+  pull_request:
   push:
+    # we need this to populate cache for `main` branch to make it available to the child branches, see
+    # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache
     branches:
       - 'main'
     tags:
       - "v*"
-  pull_request:
+  # GH caches are removed when not accessed within 7 days - this schedule runs the job every 6 days making
+  # sure that we always have some caches on master
+  schedule:
+    - cron: '0 0 */6 * *'
 
 jobs:
   build:
-    name: "Build Haskell"
-    runs-on: ${{ matrix.os }}
-    environment: cabal-cache
-
-    env:
-      # Modify this value to "invalidate" the cabal cache.
-      CABAL_CACHE_VERSION: "2024-05-01-1"
+    runs-on: ${{ matrix.sys.os }}
 
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["9.2.8", "9.6.5", "9.8.2", "9.10.1"]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        cabal: ["3.10.3.0"]
+        ghc: ["9.6", "9.8", "9.10"]
```

Comment:
we could add 9.12 now as well

Reply:
yes, it seems `cardano-cli` and `cardano-api` both have `9.12`

Reply:
Put `["9.6", "9.10", "9.12"]` like `cardano-cli`'s HEAD :+1: